lib/matplotlib/ticker.py

```diff
@@ -1212,12 +1216,32 @@ class EngFormatter(Formatter):
          24: "Y"
     }
 
-    def __init__(self, unit="", places=None):
+    def __init__(self, unit="", places=None, sep=" "):
+        """
+        Parameters
+        ----------
+        unit: string (default: "")
+            Unit symbol to use.
+
+        places: int (default: None)
+            Precision, i.e. number of digits after the decimal point.
+            If it is None, falls back to the floating point format '%g'.
+
+        sep: string (default: " ")
+            String used between the value and the prefix/unit. Beside the
+            default behavior, some other useful use cases may be:
+            * sep="" to append directly the prefix/unit to the value;
+            * sep="\\u00a0" to use a no-break space;
+            * sep="\\u202f" to use a narrow no-break space.
+        """
         self.unit = unit
         self.places = places
+        self.sep = sep
 
     def __call__(self, x, pos=None):
         s = "%s%s" % (self.format_eng(x), self.unit)
+        # Remove the trailing separator when there is neither prefix nor unit
+        s = s.strip(self.sep)
```

Comment:
This does not work the way you want. `strip` removes _any_ character in its argument from both sides: `'abcaba'.strip('aba') = 'c'`. Say someone uses `sep='0'` (for some reason), then `value = '0.00'` + ` sep = '0'` would be `'0.000'`, and it would become `'.'`.